src/test/java/com/google/javascript/clutz/partial/missing_base.d.ts

```diff
@@ -7,14 +7,22 @@ declare namespace ಠ_ಠ.clutz.module$exports$mising$extend {
   }
   class BTemplated_Instance extends ಠ_ಠ.clutz.direct.ref.ATemplated < string , number > {
   }
-  class D extends D_Instance {
+  class ClassExtendingMissingDestructuredRequire extends ClassExtendingMissingDestructuredRequire_Instance {
   }
-  //!! This emit is wrong and will be fixed with
-  //!! https://github.com/angular/clutz/issues/551.
-  class D_Instance extends ಠ_ಠ.clutz.module$contents$mising$extend_C {
+  class ClassExtendingMissingDestructuredRequire_Instance extends ಠ_ಠ.clutz.module$exports$missing$base_MissingDestructuredRequire {
```

Comment:
For case 4) mentioned above, I think this import string will not match the type string generated by a goog.provide. I suspect `ಠ_ಠ.clutz.module$exports$missing$base_MissingDestructuredRequire` will be `missing.base.MissingDestructuredRequire`. If I'm right, you might need to change the producing side to make these match.

Reply:
filed https://github.com/angular/clutz/issues/596